File: Second Year/Semester II/Web Programming/lab9_node/src/services/authenticationService.js
const userRepository = require('../repositories/userRepository');

async function validateUser(username, password) {
    const user = await userRepository.getUserByUsername(username);

    if (!user || user.password !== password) {
        return null;
    }

    return user.username;
}

module.exports = {
    validateUser
};

File: Second Year/Semester II/Web Programming/lab8_aspnet/lab8_aspnet/Services/AuthenticationService.cs
using lab8_aspnet.Repositories.Interfaces;
using lab8_aspnet.Services.Interfaces;

namespace lab8_aspnet.Services
{
    public class AuthenticationService : IAuthenticationService
    {
        private readonly IUserRepository _userRepository;

        public AuthenticationService(IUserRepository userRepository)
        {
            _userRepository = userRepository;
        }

        public async Task<string?> ValidateUserAsync(string username, string password)
        {
            var user = await _userRepository.GetUserByUsernameAsync(username);

            if (user == null || user.Password != password)
            {
                return null;
            }

            return user.Username;
        }
    }
}